src/Core/src/Platform/Android/ViewExtensions.cs

```diff
@@ -579,25 +580,7 @@ internal static void Arrange(this IView view, View platformView)
 			=> GetHostedWindow(view?.Handler?.PlatformView as View);
 
 		internal static IWindow? GetHostedWindow(this View? view)
-			=> GetWindowFromActivity(view?.Context?.GetActivity());
-
-		internal static IWindow? GetWindowFromActivity(this Android.App.Activity? activity)
-		{
-			if (activity is null)
-				return null;
-
-			var windows = WindowExtensions.GetWindows();
-			foreach (var window in windows)
-			{
-				if (window.Handler?.PlatformView is Android.App.Activity active)
-				{
-					if (active == activity)
-						return window;
-				}
-			}
-
-			return null;
-		}
+			=> view?.Context?.GetWindow();
```

Comment:
Ha you were already consolidating. 

XF: yea it's really hard to get the window
MAUI: here's like 7 extensions you can use with slightly different names :-p